tests/test_kaggle_module_resolver.py

```diff
@@ -0,0 +1,66 @@
+import unittest
+
+import os
+import threading
+
+import tensorflow as tf
+import tensorflow_hub as hub
+
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from test.support import EnvironmentVarGuard
+
+
+class TestKaggleModuleResolver(unittest.TestCase):
+    class HubHTTPHandler(BaseHTTPRequestHandler):
+        def do_GET(self):
+            print("GETGETGET")
+            self.send_response(200)
+            self.send_header('Content-Type', 'application/gzip')
+            self.end_headers()
+
+            with open('/input/tests/data/model.tar.gz', 'rb') as model_archive:
+                self.wfile.write(model_archive.read())
+
+    def _test_client(self, client_func, handler):
+        with HTTPServer(('localhost', 8080), handler) as test_server:
+            threading.Thread(target=test_server.serve_forever).start()
+
+            try:
+                client_func()            
+            finally:
+                test_server.shutdown()
+
+    def test_kaggle_resolver_succeeds(self):
+        # Simulates we are inside a Kaggle environment.
+        env = EnvironmentVarGuard()
+        env.set('KAGGLE_CONTAINER_NAME', 'foo')
+        # Attach model to right directory.
+        os.makedirs('/kaggle/input/foomodule/tensorflow2/barvar')
```

Comment:
Shouldn't it have the `VersionNumber` at the end?

Reply:
I symlink the directory at the `VersionNumber` the line below.